src/Monolog/Handler/BrowserConsoleHandler.php

```diff
@@ -131,21 +145,27 @@ protected static function getResponseFormat(): string
         // Check content type
         foreach (headers_list() as $header) {
             if (stripos($header, 'content-type:') === 0) {
-                // This handler only works with HTML and javascript outputs
-                // text/javascript is obsolete in favour of application/javascript, but still used
-                if (stripos($header, 'application/javascript') !== false || stripos($header, 'text/javascript') !== false) {
-                    return 'js';
-                }
-                if (stripos($header, 'text/html') === false) {
-                    return 'unknown';
-                }
-                break;
+                return static::getResponseFormatFromContentType($header);
             }
         }
 
         return 'html';
     }
 
+    protected static function getResponseFormatFromContentType(string $contentType): ?string
+    {
+        // This handler only works with HTML and javascript outputs
+        // text/javascript is obsolete in favour of application/javascript, but still used
+        if (stripos($contentType, 'application/javascript') !== false || stripos($contentType, 'text/javascript') !== false) {
+            return static::FORMAT_JS;
+        }
+        if (stripos($contentType, 'text/html') === false) {
+            return null;
```

Comment:
I do question the value of doing this change to `null`.. why not keep `'unknown'`, so it always returns a string? Right now getResponseFormat's return value is broken for example as you didn't update it.

I don't really see the value in deprecating this, except causing more fragmentation in implementations, and added work.